Sources/StructuralCore/Structural.swift

```diff
@@ -12,7 +12,60 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/// A type that can be converted to and from its structural representation.
+/// Types that can be converted to and from their structural representations.
+///
+/// The compiler can synthesize conformances to the `Structural` protocol for
+/// `struct` and `enum` declarations.
+///
+/// For `struct` declarations, the canonical structural representation type is
+/// `StructuralStruct<...>`, whose generic parameter contains information
+/// representing all stored properties.
+///
+/// `struct` example:
+///
+///     struct StudentGrades {
+///         let classId: Int
+///         var grades: [Double]
+///     }
+///
+///     extension StudentGrades: Structural {
+///         typealias StructuralRepresentation = StructuralStruct<
+///             StructuralCons<
+///                 StructuralProperty<Int>,
+///                 StructuralCons<
+///                     StructuralProperty<[Double]>,
+///                     StructuralEmpty
+///                 >
+///             >
+///         >
+///     }
+///
+/// For `enum` declarations, the canonical structural representation type is
+/// `StructuralEnum<...>`, whose generic parameter contains information
+/// representing all enum cases and associated values.
+///
+/// `enum` example:
+///
+///     enum Color: Int {
+///         case red = 0xFF0000
+///         case green = 0x00FF00
+///         case blue = 0x0000FF
+///     }
+///
+///     extension Color: Structural {
+///         // Compiler synthesizes:
+///         typealias StructuralRepresentation =
+///             StructuralEnum<
```

Comment:
`StructuralCons` and `StructuralEither` are both type-level heterogeneous lists.

`StructuralCons` (for struct stored properties and enum associated values) ends with a `StructuralEmpty` empty list base case.

However, `StructuralEither` seems to have no such base case. **Should it (also use `StructuralEmpty`)?**

This seems necessary for defining `StructuralRepresentation` for an empty enum (`StructuralEnum<StructuralEmpty>`).

Reply:
Thanks for pointing this out, it's indeed seems to be inconsistent! Feel free to address this in a follow-up PR (examples protocols & structurs might have to be updated to fully validate this direction).